app/ui-message/client/blocks/MessageBlock.js

```diff
@@ -74,11 +69,11 @@ export const modalBlockWithContext = ({
 			const element = ref.current.querySelector(focusableElementsString);
 			element && element.focus();
 		}
-	}, [ref.current, data.errors]);
+	}, [data]);
 	// save focus to restore after close
 	const previousFocus = useMemo(() => document.activeElement, []);
 	// restore the focus after the component unmount
-	useEffect(() => () => previousFocus && previousFocus.focus(), []);
+	useEffect(() => () => previousFocus && previousFocus.focus(), [previousFocus]);
```

Comment:
since all the modal rerenders again... any change in any file will focus the first element